### PR TITLE
PLT-4591 [FE] - Add support for auth proof from the client of the private signature from the wallet

### PR DESCRIPTION
**Pull request by amnambiar**
https://input-output.atlassian.net/browse/PLT-4591

### CODE REVIEW

File: react-web/src/components/ConnectWallet/ConnectWallet.tsx [hunk 1]
```diff
@@ -42,21 +47,64 @@ const ConnectWallet = () => {
...
+                        catchError(err)
+                    })
+                }
+
+            } catch (err) {
+                catchError(err)
+            }
+        }).catch((err) => {
+            catchError(err)
+        })
```
Comment (bogdan-manole) [inline react-web/src/components/ConnectWallet/ConnectWallet.tsx:107]: - Why not using async/await
- The code is not clean and hard to follow with all these multiple catches. Here is an alternative: 
```suggestion
    const initiatePrivateWalletSignature = async (currentWallet: any, walletAddr_bech32: any, walletAddr: string) => {
      try{
        const timestamp = (await fetchData.get<any,AxiosResponse<number>,any>('/server-timestamp')).data;
        const msgToBeSigned = `Sign this message if you are the owner of the ${walletAddr_bech32} address. \n Timestamp: <<${timestamp}>> \n Expiry: 60 seconds`;
        const {key, signature} = await currentWallet.signData(walletAddr, Buffer.from(msgToBeSigned, 'utf8').toString('hex'))
        if (key && signature) {
          const token = (await fetchData.post<any,AxiosResponse<string>,any>('/login', {
            address: walletAddr_bech32,
            key: key,
            signature: signature
          })).data;
          localStorage.setItem('authToken', token)
          setAddress(walletAddr_bech32)
        }
      } catch (err) {
        catchError(err)
      }
```
- If you agree with the alternative there is still an issue: What do we do when there is no (key/signature), I see you have an if, but there is no else branch, shouldn't the user be notified somehow?

**Review by bogdan-manole — CHANGES_REQUESTED** [review]
Small changes, and some refactoring, bot other than this, looks good to me

File: react-web/src/components/ConnectWallet/ConnectWallet.tsx [hunk 2]
```diff
@@ -1,14 +1,15 @@
 import React, { useEffect, useState, useCallback } from "react";
 import { Address } from "@emurgo/cardano-serialization-lib-browser";
 import { useAppDispatch } from "store/store";
-import { getProfileDetails, setNetwork } from "store/slices/auth.slice";
+import { getProfileDetails, logout, setNetwork } from "store/slices/auth.slice";
 
 import Modal from "components/Modal/Modal";
 import Button from "components/Button/Button";
 import Loader from "components/Loader/Loader";
 import Toast from "components/Toast/Toast";
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply react-web/src/components/ConnectWallet/ConnectWallet.tsx:9]: It was used at LoC 78 for `dispatch(logout())`
Instead removed unused Toast at LoC 9

File: react-web/src/components/ConnectWallet/ConnectWallet.tsx (hunk 1 above)
Comment (amnambiar) [reply react-web/src/components/ConnectWallet/ConnectWallet.tsx]: Done
